src/mongo/db/audit/audit.cpp

```diff
@@ -214,6 +227,101 @@ namespace audit {
         SimpleMutex _mutex;
     };    
 
+    // Writes audit events to a json file
+    class JSONAuditLog: public FileAuditLog {
+
+        class Adapter: public AuditLogFormatAdapter {
+            const std::string str;
+
+        public:
+            Adapter(const BSONObj &obj)
+                : str(mongoutils::str::stream() << obj.jsonString() << "\n") {}
```

Comment:
Neglect my previous comment, I thought you're appending data somewhere later to this variable...